db/migrate/20241229165332_add_devise_to_users.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[8.0]
+  def change
+    rename_column :users, :crypted, :legacy_password_hash
+    change_column_null :users, :legacy_password_hash, true
```

Comment:
This is awkward for a reversible migration. Maybe we should have a temporary period where we still fill in the `legacy_password_hash` in addition to Devise, so we can down-migrate more easily? I'm not sure where to override to do that, though; would need to investigate.

Reply:
Updated to reset passwords to a random string on migrate-down if they've already been migrated. It'll require the relevant users to perform a password reset, but it's better than making this migration irreversible, in case something does go wrong with the deploy!